Src/FluentAssertions/Equivalency/IValidateNestedEquivalency.cs

```diff
@@ -0,0 +1,7 @@
+namespace FluentAssertions.Equivalency
+{
+    public interface IValidateNestedEquivalency
+    {
+        void RecursivelyAssertEquality(IEquivalencyValidationContext context);
+    }
```

Comment:
I'm not sure I see the use cases or need for specifying `Nested` and `Recursively` 🤔
`Nested` is a part of how we _define_ equivalency.
`Recursively` is an implementation detail of how we _verify_ equivalency.

Reply:
During several of my masterclasses, the students mentioned that it was not obvious that this method would be called recursively. By adding this to the name, it would be enough of a hint to think about the consequences.

Reply:
Playing the devil's advocate:
Why do you need to know it's recursive?
If I call a `Sort` function on sequence I don't know whether it's recursively or iteratively implemented, only that upon return I have a sorted sequence.

brain storming some names:
* `ValidateEquivalency`
* `Validate` (since the Equivalency part is already part of the interface name)

Reply:
You need to make sure it's re-entrant. But I fail to come up with a good example of that right now..